kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -320,6 +340,36 @@ public String toString() {
     }
   }
 
+  private static class CannedAclValidator implements ConfigDef.Validator {
+    public static Map<String, CannedAccessControlList> aclsByHeaderValue = new HashMap<>();
+    private static String allowedValues;
+
+    static {
+      List<String> aclHeaderValues = new ArrayList<>();
+      for (CannedAccessControlList acl : CannedAccessControlList.values()) {
+        aclsByHeaderValue.put(acl.toString(), acl);
+        aclHeaderValues.add(acl.toString());
+      }
+      allowedValues = Utils.join(aclHeaderValues, ", ");
+    }
+
+    @Override
+    public void ensureValid(String name, Object cannedAcl) {
+      if (cannedAcl == null) {
+        return;
+      }
```

Comment:
Let me know if you'd prefer a style other than an early return for the null case.